editor/src/clj/editor/code/view.clj

```diff
@@ -3030,31 +2972,40 @@
   ;; background process has modified (or even deleted) the file
   ;; without the editor knowing, the "original" unmodified lines
   ;; reached after a series of undo's could be something else entirely
   ;; than what the user saw.
-  (g/with-auto-evaluation-context ec
-    (r/ensure-loaded! resource-node ec)
-    (let [glyph-metrics (g/node-value view-node :glyph-metrics ec)
-          tab-spaces (g/node-value view-node :tab-spaces ec)
-          tab-stops (data/tab-stops glyph-metrics tab-spaces)
-          lines (g/node-value resource-node :lines ec)
-          document-width (data/max-line-width glyph-metrics tab-stops lines)
-          resource (g/node-value resource-node :resource ec)]
-      (when (resource/file-resource? resource)
-        (lsp/open-view! lsp view-node resource lines))
+  (let [resource (resource-node/resource resource-node)
+        resource-type (resource/resource-type resource)
+        resource-node-type (:node-type resource-type)
+        is-code-resource-type (r/code-resource-type? resource-type)]
+    (when is-code-resource-type
+      (g/with-auto-evaluation-context evaluation-context
+        (r/ensure-loaded! resource-node evaluation-context)))
+    (let [[lines document-width]
+          (g/with-auto-evaluation-context evaluation-context
+            (let [glyph-metrics (g/node-value view-node :glyph-metrics evaluation-context)
+                  tab-spaces (g/node-value view-node :tab-spaces evaluation-context)
+                  tab-stops (data/tab-stops glyph-metrics tab-spaces)
+                  lines (g/node-value resource-node :lines evaluation-context)
+                  document-width (data/max-line-width glyph-metrics tab-stops lines)]
+              [lines document-width]))]
       (g/transact
         (concat
           (g/set-property view-node :document-width document-width)
-          (g/connect resource-node :completions view-node :completions)
-          (g/connect resource-node :cursor-ranges view-node :cursor-ranges)
-          (g/connect resource-node :indent-type view-node :indent-type)
-          (g/connect resource-node :invalidated-rows view-node :invalidated-rows)
-          (g/connect resource-node :lines view-node :lines)
-          (g/connect resource-node :regions view-node :regions)
-          (g/connect app-view :debugger-execution-locations view-node :debugger-execution-locations)))
+          (g/connect app-view :debugger-execution-locations view-node :debugger-execution-locations)
+          (gu/connect-existing-outputs resource-node-type resource-node view-node
+            [[:completions :completions]
+             [:cursor-ranges :cursor-ranges]
+             [:indent-type :indent-type]
+             [:invalidated-rows :invalidated-rows]
+             [:lines :lines]
+             [:regions :regions]])))
+      (when (and is-code-resource-type
+                 (resource/file-resource? resource))
+        (lsp/open-view! lsp view-node resource lines))
```

Comment:
Don't bother with Language Server stuff unless this is a code file, and only connect outputs that exist. For non-`CodeEditorResourceNodes`, that's only really the `lines` output.